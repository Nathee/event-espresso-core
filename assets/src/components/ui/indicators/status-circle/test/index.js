/**
 * External imports
 */
import { shallow } from 'enzyme';

/**
 * Internal imports
 */
import StatusCircle, { statusSizes } from '../index';


describe( 'StatusCircle', () => {
	it( 'renders as expected with default props', () => {
		const wrapper = shallow( <StatusCircle statusCode={ 'TFL' } /> );
		expect( wrapper ).toMatchSnapshot();
	} );
	it( 'renders expected class names for different size passed in', () => {
		const wrapper = shallow(
			<StatusCircle
				statusCode={ 'TFL' }
				circleSize={ statusSizes[ 32 ] }
			/>
		);
		expect(
			wrapper
				.find( '.ee-status-circle' )
				.hasClass( 'status-bg-TFL circle-size-32' )
		).toBe( true );
	} );
	it( 'renders custom class names passed in', () => {
		const wrapper = shallow(
			<StatusCircle
				statusCode={ 'TFL' }
				className={ 'customa customb' }
			/>
		);
		expect(
			wrapper
				.find( '.ee-status-circle' )
				.hasClass( 'customa customb' )
		).toBe( true );
	} );
} );