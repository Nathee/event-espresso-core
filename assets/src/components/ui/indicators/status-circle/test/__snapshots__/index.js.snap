// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusCircle renders as expected with default props 1`] = `
<span
  className="ee-status-circle status-bg-TFL circle-size-12"
/>
`;
